sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -57,4 +74,20 @@ public void evaluateAssign(Consumer consumer, Collection<TopicPartition> topicPa
     mapContext.setVariable("tp", topicPartitions);
     assignExpression.getValue(mapContext);
   }
+
+  public long getRecordTimestamp(ConsumerRecord<byte[], byte[]> rawRecord) {
+    long timestamp = -1L;
+    try {
+      //for Kafka 0.9, set to System.currentTimeMillis();
+      //for kafka 0.10, when NO_TIMESTAMP also set to System.currentTimeMillis();
+      if (!hasRecordTimestamp || (timestamp = (long) timestampMethod.invoke(rawRecord)) == -1L) {
```

Comment:
`-1L` is the timestamp returned when `NO_TIMESTAMP` in Kafka0.10, could change to `<=0` to cover more unexpected cases.